q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,312 @@
+    return False
+
+
+def check_validation(rule_value):
+    """Check that the user-defined value for the
+    rule "validation" is correctly formatted.
+
+    validation must be a dict
+        Apply the rule of validating variables values: replace by the blank
+        (or the missing) value every sample entry that for the current variable has
+        in one or multiple other variable(s) a null or missing value for that sample.
+                    validation:
+                      force_to_blank_if:
+                        is null:
+                        - variable
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "validation".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+
+    """
+    # accepted structure for the "validation" rule
+    possible_validations = {
+        'force_to_blank_if': {'is null'}
+    }
+    # check the first level is a dict
+    if isinstance(rule_value, dict):
+        # get the current keys
+        rule_value_set = set(rule_value)
+        # get the accepted keys from the current keys
+        accepted_set = rule_value_set & set(possible_validations)
+        # if there is not more keys than the accepted ones
+        if not len(rule_value_set ^ accepted_set):
+            # get the possible conditions
+            possible_validations_2 = possible_validations['force_to_blank_if']
+            # for each of the condition/replacements for the accepted keys
+            for rule_2, rule_value_2 in rule_value['force_to_blank_if'].items():
+                # check that the content of replacements are lists
+                if isinstance(rule_value_2, list):
+                    # get the accepted conditions from the current conditions
+                    if rule_2 in possible_validations_2:
+                        return True
+    return False
+
+
+def check_normalization(rule_value):
+    """Check that the user-defined value for the
+    rule "normalization" is correctly formatted.
+
+    normalization must be a dict
+        Apply the rules of normalizing the variables values, by replacing
+        the values outside the range by a given replacement value.
+                    normalization:
+                        gated_value:
+                            - Out of bounds
+                        maximum:
+                            - int
+                            - float
+                        minimum:
+                            - int
+                            - float
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "normalization".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+
+    """
+    if isinstance(rule_value, dict):
+        possible_normalizations = {'maximum', 'minimum', 'gated_value'}
+        rule_value_set = set(rule_value)
+        if not len(rule_value_set ^ (rule_value_set & possible_normalizations)):
```

Comment:
If you have a set max or min, should be allowed to not have a gated value? Or set default gated value?

Reply:
@ElDeveloper / @wasade that's a good question:
Shall a value that is - per this rule - found either above the given max. or below the given min. be replaced by `np.nan` by default in case no `gated_value` is provided? and/or shall the tool raise a warning such as **"no gated value privided for variable _x_: using default NaN"**?

Reply:
My understanding was that values outside of min/max were errors, right?

Reply:
Yes, according to the rule, for example one may set a maximum to 120 years-old (while there might be really someone 121 years olds)... or for another example, one may set the rule to 60 in case you want to then only have one big group for >60, in this case, the rules, would be:
maximum: "60"
gated_value: ">60"

But in both cases, the problem is not really what the values outside of min/max represent (i.e. errors or user-unwanted), but rather what the values outside of min/max should be replaced by in the return metadata file.

This summarizes Rebecca-and-myself's wonder: would we allow the "gated_value:" rule key to be missing? and it this case, what would we use as default return value for a then-"error" value: np.nan, controlled-vacabulary "Not provided", "Not collected" or "Missing: not applicable", something else...?

(in the same vein, note that a user passing no "minimum" (or no "maximum") in the rule keys could actually be okay, ... and then the actual minimum under the hood would be set to None, i..e no normalization.. that sounds good?)

Reply:
Thanks for breaking this down Franck.

> would we allow the "gated_value:" rule key to be missing?

The most conservative approach here would be to require a `gated_value`  attribute to avoid situations where user's data is "accidentally" lost during normalization. Suggesting a default value of "Missing: not applicable" would probably make sense.

Reply:
Agreed. In this case, this code for function `check_normalization` is okay as is. Missing cases of `minimum`, `maximum` or `gated_value` will be handled as the rule applies.